setup.py

```diff
@@ -55,10 +55,12 @@
         'License :: OSI Approved :: Apache Software License',
         'Natural Language :: English',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8'
+        'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12'
     ],
-    python_requires=">=3.7",
+    python_requires=">=3.12",
```

Comment:
We need to support all active versions of Python so this should be 3.9 and greater.

Reply:
Yes, I chaged the python version to >=3.9. Also the changes are tested with Python version 3.9, 3.10, 3.11, 3.12 which are supported.